src/asset/asset-list-loader.js

```diff
@@ -0,0 +1,214 @@
+import { EventHandler } from '../core/event-handler.js';
+
+import { Asset } from './asset.js';
+
+/**
+ * @class
+ * @name AssetListLoader
+ * @augments EventHandler
+ * @classdesc Used to load a group of assets and fires a callback when all assets are loaded.
+ * @param {Asset[]|number[]} assetList - An array of {@link Asset} objects to load or an array of Asset IDs to load.
+ * @param {AssetRegistry} assetRegistry - The application's asset registry.
+ */
+class AssetListLoader extends EventHandler {
+    constructor(assetList, assetRegistry) {
+        super();
+        this._assets = [];
+        this._loadingAssets = new Set();
+        this._registry = assetRegistry;
+        this._loaded = false;
+        this._count = 0; // running count of successfully loaded assets
+        this._failed = []; // list of assets that failed to load
+
+        this._waitingAssets = [];
+
+        if (assetList.length && assetList[0] instanceof Asset) {
+            assetList.forEach((asset, i, array) => {
+                // filter out duplicates
+                if (array.indexOf(asset) !== i) {
+                    return;
+                }
+                if (!asset.registry) {
+                    asset.registry = assetRegistry;
+                }
+                this._assets.push(asset);
+            });
+        } else {
+            // list of Asset IDs
+            assetList.forEach((assetId, i, array) => {
+                // filter out duplicates
+                if (array.indexOf(assetId) !== i) {
+                    return;
+                }
+                var asset = assetRegistry.get(assetList[i]);
+                if (asset) {
+                    this._assets.push(asset);
+                } else {
+                    this._waitForAsset(assetList[i]);
+                }
+            });
+        }
+    }
+
+    destroy() {
```

Comment:
Should this be public API?

Reply:
I'm not sure why this change isn't showing up but it's been added: https://github.com/playcanvas/engine/pull/4149/commits/fb1b9b088b656cf815626df121582a83896815ff